buildSrc/src/main/kotlin/gradle/plugins/DataModule.kt

```diff
@@ -0,0 +1,47 @@
+package gradle.plugins
+
+import annotationProcessors
+import com.android.build.gradle.LibraryExtension
+import implementationDependenciesFrom
+import implementationProjectsFrom
+import org.gradle.api.Project
+import org.gradle.kotlin.dsl.apply
+import org.gradle.kotlin.dsl.getByType
+
+class DataModule : BaseAndroidPlugin() {
+    override fun apply(project: Project) {
+        appPlugins(project)
+        super.apply(project)
+
+        dependencies(project)
+    }
+
+    private fun dependencies(project: Project) {
+        project.implementationProjectsFrom(
+            listOf(
+                ":domain",
+                ":common",
+            )
+        )
+
+        project.implementationDependenciesFrom(
+            roomImplDeps + hiltImplDeps + Lib.Kotlin.androidList + Lib.Kotlin.kotlinList + Lib.Networking.ktorList
+        )
+
+        project.annotationProcessors(
+            hiltKapt + listOf(Lib.Room.roomCompiler)
+        )
+
+        project.extensions.getByType<LibraryExtension>().apply {
+            configureAndroidLibraryBlock(project)
+        }
```

Comment:
Do these lines of code contain all of the following?
```
android {
  compileSdk = 31

  defaultConfig {
    minSdk = (AppVersions.MIN_SDK)
    targetSdk = (AppVersions.TARGET_SDK)
    testInstrumentationRunner = "androidx.test.runner.AndroidJUnitRunner"
  }

  buildTypes {
    getByName("release") {
      isMinifyEnabled = false
      proguardFiles(getDefaultProguardFile("proguard-android-optimize.txt"), "proguard-rules.pro")
    }
  }
```

Reply:
Yeah @shubhamsinghmutualmobile 
same function as made by @anmolvermamm 

fun LibraryExtension.configureAndroidLibraryBlock(project: Project) {
  compileSdkVersion = AppVersions.COMPILE_SDK

  defaultConfig {
    minSdk = (AppVersions.MIN_SDK)
    targetSdk = (AppVersions.TARGET_SDK)
    testInstrumentationRunner = "androidx.test.runner.AndroidJUnitRunner"
  }

  buildTypes {
    getByName("release") {
      isMinifyEnabled = false
      proguardFiles(getDefaultProguardFile("proguard-android-optimize.txt"), "proguard-rules.pro")
    }
  }

  buildFeatures.compose = true

  composeOptions {
    kotlinCompilerExtensionVersion = Lib.Androidx.composeVersion
  }
  packagingOptions {
    resources.excludes.add("META-INF/LICENSE.txt")
    resources.excludes.add("META-INF/NOTICE.txt")
    resources.excludes.add("LICENSE.txt")
    resources.excludes.add("/META-INF/{AL2.0,LGPL2.1}")
  }

  compileOptions {
    sourceCompatibility = JavaVersion.VERSION_1_8
    targetCompatibility = JavaVersion.VERSION_1_8
  }
  project.tasks.withType<KotlinCompile>().configureEach {
    kotlinOptions {
      jvmTarget = JavaVersion.VERSION_1_8.toString()
    }
  }
}